CMakeLists.txt

```diff
@@ -7,34 +7,14 @@ if (CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")
   set(CMAKE_SYSTEM_PROCESSOR "amd64")
 endif()
 
-#----------------------------------------------------------------------
-# Use the grpc targets directly from this build, only when not cross-compiling.
-#----------------------------------------------------------------------
-if(CMAKE_CROSSCOMPILING)
-  find_program(_PROTOBUF_PROTOC protoc)
-  find_program(_GRPC_CPP_PLUGIN_EXECUTABLE grpc_cpp_plugin)
-
-  if(NOT _GRPC_DEVICE_NILRT_LEGACY_TOOLCHAIN)
-    find_package(gRPC REQUIRED)
-    find_library(_REFLECTION grpc++_reflection)
-    find_library(_GRPC_GRPCPP grpc++)
-    find_library(_PROTOBUF_LIBPROTOBUF protobuf)
-  else()
-    add_subdirectory(third_party/grpc ${CMAKE_CURRENT_BINARY_DIR}/grpc EXCLUDE_FROM_ALL)
-    set(_REFLECTION grpc++_reflection)
-    set(_GRPC_GRPCPP grpc++)
-    set(_PROTOBUF_LIBPROTOBUF libprotobuf)
-  endif()
 
-else()
-  add_subdirectory(third_party/grpc ${CMAKE_CURRENT_BINARY_DIR}/grpc EXCLUDE_FROM_ALL)
-  set(_PROTOBUF_PROTOC $<TARGET_FILE:protobuf::protoc>)
-  set(_REFLECTION grpc++_reflection)
-  set(_GRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:grpc_cpp_plugin>)
-  set(_GRPC_GRPCPP grpc++)
-  set(_PROTOBUF_LIBPROTOBUF libprotobuf)
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+if(NOT CMAKE_CROSSCOMPILING OR _GRPC_DEVICE_NILRT_LEGACY_TOOLCHAIN)
+  add_subdirectory(${CMAKE_SOURCE_DIR}/third_party/grpc ${CMAKE_CURRENT_BINARY_DIR}/grpc EXCLUDE_FROM_ALL)
```

Comment:
Is the `${CMAKE_SOURCE_DIR}` fully necessary here? If so, consider using `${CMAKE_CURRENT_SOURCE_DIR}` instead. While they should be the same in a top-level CMakeLists.txt file, it should ensure that it evaluates correctly if this is ever run from another project.

Reply:
Oh, hm. Thanks for bringing this up.

Yeah, these are not needed. I added some of that when I was trying to move the add_subdirectory into ImportGrpc, but that didn't work.

Pragmatically, it's easiest to use `${CMAKE_SOURCE_DIR}` to reference files in fixed locations off of the root. When it's in a cmake script that is shared between multiple cmakelists, I'm not sure there's an easy alternative (since they each may be in different relative locations to the root). But that does prevent building them as a separate top-level projects. Since, that's not specifically a goal of this effort, I think I'll call that "out of scope" and stick with `${CMAKE_SOURCE_DIRECTORY}` in `cmake` scripts. 

In `CMakeLists.txt` files it make sense to use `{CMAKE_CURRENT_SOURCE_DIR}` since you have a fixed starting location (unless you move the file ...), Although combining it with `../../` is kind of clunky and I don't think that's adding any real portability. So, I'd probably prefer `${CMAKE_SOURCE_DIR}` for accessing fixed dirs off of the root there as well.